libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -393,13 +394,37 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 
 }
 
-void DeclarationTypeChecker::endVisit(UsingForDirective const& _usingFor)
+bool DeclarationTypeChecker::visit(UsingForDirective const& _usingFor)
 {
 	ContractDefinition const* library = dynamic_cast<ContractDefinition const*>(
 		_usingFor.libraryName().annotation().referencedDeclaration
 	);
+
 	if (!library || !library->isLibrary())
 		m_errorReporter.fatalTypeError(4357_error, _usingFor.libraryName().location(), "Library name expected.");
+
+	_usingFor.libraryName().annotation().type = TypeProvider::contract(*library);
+	if (_usingFor.typeName())
+		_usingFor.typeName()->accept(*this);
+
+	return false;
+}
+
+bool DeclarationTypeChecker::visit(InheritanceSpecifier const& _inheritanceSpecifier)
+{
+	if (auto const* contract = dynamic_cast<ContractDefinition const*>(
+		_inheritanceSpecifier.name().annotation().referencedDeclaration
+	))
```

Comment:
Shouldn't this `if` rather be an assertion?

Reply:
Indeed the type checker asserts for this, so I guess it's fine to do it here, too, yeah.